migrations/20200824194331-patient-gender.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction((t) => {
+      return Promise.all([
+        queryInterface.bulkUpdate(
+          'patients',
+          { gender: 'M' },
+          { gender: 'Male' },
+          { transaction: t }
+        ),
+        queryInterface.bulkUpdate(
+          'patients',
+          { gender: 'F' },
+          { gender: 'Female' },
+          { transaction: t }
+        ),
+        queryInterface.changeColumn(
+          'patients',
+          'gender',
+          {
+            type: Sequelize.DataTypes.ENUM('M', 'F'),
+          },
+          {
+            transaction: t,
+          }
+        ),
+      ]);
+    });
+  },
```

Comment:
two issues here:
(1) we don't handle values that aren't `Male` or `Female` here - this means that when we try to do the `changeColumn` operation, any invalid values are still in the table and causes it to error out; for simplicity's sake, let's just take all these values and convert them into `F`

(2) we need the updates to complete first before attempting to changing the column type, otherwise it'll error out if there are any invalid values due to race conditions - transactions don't ensure the order of operations is preserved

the following suggestion only attempts to address (2) and should be tested as well
```suggestion
  up: (queryInterface, Sequelize) => {
    return queryInterface.sequelize.transaction((t) => {
      return Promise.all([
        queryInterface.bulkUpdate(
          'patients',
          { gender: 'M' },
          { gender: 'Male' },
          { transaction: t }
        ),
        queryInterface.bulkUpdate(
          'patients',
          { gender: 'F' },
          { gender: 'Female' },
          { transaction: t }
        ),
      ]).then(() =>
        queryInterface.changeColumn(
          'patients',
          'gender',
          {
            type: Sequelize.DataTypes.ENUM('M', 'F'),
          },
          {
            transaction: t,
          }
        )
      );
    });
  },
```

Reply:
For condition 1, is this the expected behaviour you wanted?
![Screen Shot 2020-08-26 at 1 06 49 PM](https://user-images.githubusercontent.com/21224282/91345989-76f6f800-e79d-11ea-8937-d0484a39d879.png)

Reply:
@aaronabraham311 not quite - it shouldn't end up with all three as `F`; end results should be something like:
`Male` -> `M`
`Female` -> `M`
Other -> `F`